### DISCUSSION TITLE
Setting a multiline environment variable fails

### DISCUSSION THREAD

**Discussion by fpoli** [community/community#24952] [Actions]
<p>I tried to set a multiline environment variable from a Github action following the <a href="https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#multiline-strings" rel="noopener nofollow ugc">documentation</a>, but I get an error that I can’t reproduce locally. Do you have any clue about the reason, or suggestions on how to debug this? Ideally, I would like a docker image that does a dry run of the workflow.</p>
<pre><code class="lang-bash"># Find outdated dependencies
echo 'OUTDATED_DEPENDENCIES=&lt;&lt;EOF' &gt;&gt; $GITHUB_ENV
cargo outdated --root-deps-only --workspace &gt;&gt; $GITHUB_ENV
echo 'EOF' &gt;&gt; $GITHUB_ENV
</code></pre>
<p>Error:</p>
<pre><code class="lang-auto">Unable to process file command 'env' successfully.
Invalid environment variable format 'analysis'
</code></pre>
<p>“analysis” is indeed the first word that I want to store in the environment variable.</p>
<details>
<summary>Output of the `cargo outdated ...` command</summary>
<pre><code class="lang-nohighlight">analysis
================
Name            Project  Compat  Latest  Kind         Platform
----            -------  ------  ------  ----         --------
compiletest_rs  0.5.0    ---     0.6.0   Development  ---
serde_json      1.0.61   1.0.64  1.0.64  Normal       ---

prusti
================
Name        Project  Compat  Latest  Kind    Platform
----        -------  ------  ------  ----    --------
env_logger  0.8.2    0.8.3   0.8.3   Normal  ---

prusti-common
================
Name    Project  Compat  Latest  Kind    Platform
----    -------  ------  ------  ----    --------
config  0.9.3    ---     0.10.1  Normal  ---

viper
================
Name        Project  Compat  Latest  Kind         Platform
----        -------  ------  ------  ----         --------
env_logger  0.8.2    0.8.3   0.8.3   Development  ---
jni         0.17.0   ---     0.19.0  Normal       ---

viper-sys
================
Name        Project  Compat  Latest  Kind    Platform
----        -------  ------  ------  ----    --------
env_logger  0.8.2    0.8.3   0.8.3   Normal  ---
jni         0.17.0   ---     0.19.0  Normal  ---
ureq        1.5.2    1.5.4   2.0.2   Build   ---

jni-gen
================
Name  Project  Compat  Latest  Kind    Platform
----  -------  ------  ------  ----    --------
jni   0.17.0   ---     0.19.0  Normal  ---

prusti-contracts
================
Name      Project  Compat  Latest  Kind         Platform
----      -------  ------  ------  ----         --------
trybuild  1.0.40   1.0.41  1.0.41  Development  ---

prusti-contracts-impl
================
Name   Project  Compat  Latest  Kind    Platform
----   -------  ------  ------  ----    --------
quote  1.0.8    1.0.9   1.0.9   Normal  ---

prusti-specs
================
Name        Project  Compat  Latest  Kind    Platform
----        -------  ------  ------  ----    --------
quote       1.0.8    1.0.9   1.0.9   Normal  ---
serde_json  1.0.61   1.0.64  1.0.64  Normal  ---

prusti-interface
================
Name    Project  Compat  Latest  Kind    Platform
----    -------  ------  ------  ----    --------
config  0.9.3    ---     0.10.1  Normal  ---

prusti-server
================
Name        Project  Compat  Latest  Kind    Platform
----        -------  ------  ------  ----    --------
bincode     1.3.1    1.3.2   1.3.2   Normal  ---
env_logger  0.8.2    0.8.3   0.8.3   Normal  ---
futures     0.1.29   0.1.31  0.3.13  Normal  ---
reqwest     0.9.24   ---     0.11.1  Normal  ---
tokio       0.1.22   ---     1.2.0   Normal  ---
warp        0.1.23   ---     0.3.0   Normal  ---

prusti-tests
================
Name            Project  Compat  Latest  Kind         Platform
----            -------  ------  ------  ----         --------
compiletest_rs  0.5.0    ---     0.6.0   Development  ---

prusti-launch
================
Name   Project  Compat  Latest  Kind    Platform
----   -------  ------  ------  ----    --------
ctrlc  3.1.7    3.1.8   3.1.8   Normal  ---
nix    0.19.1   ---     0.20.0  Normal  cfg(unix)

test-crates
================
Name        Project  Compat  Latest  Kind    Platform
----        -------  ------  ------  ----    --------
env_logger  0.8.2    0.8.3   0.8.3   Normal  ---
rustwide    0.11.0   0.11.1  0.12.0  Normal  ---

systest
================
Name        Project  Compat  Latest  Kind    Platform
----        -------  ------  ------  ----    --------
env_logger  0.8.2    0.8.3   0.8.3   Build   ---
jni         0.17.0   ---     0.19.0  Normal  ---
ureq        1.5.2    1.5.4   2.0.2   Build   ---
</code></pre>
</details>
<p>Github action run: <a href="https://github.com/viperproject/prusti-dev/actions/runs/610277666" rel="noopener nofollow ugc">here</a></p>
<details>
<summary>Workflow file:</summary>
<p>Link: <a href="https://github.com/viperproject/prusti-dev/blob/84300ba40d022aed12c803c4840386e00712ae2c/.github/workflows/update-deps.yml" class="inline-onebox" rel="noopener nofollow ugc">prusti-dev/update-deps.yml at 84300ba40d022aed12c803c4840386e00712ae2c · viperproject/prusti-dev · GitHub</a></p>
<pre><code class="lang-yaml">name: Update dependencies

on:
  workflow_dispatch:
  schedule:
    # At 07:00 UTC on day-of-month 1 and 15. Use https://crontab.guru/ to edit this.
    - cron:  '0 7 1,15 * *'

jobs:
  # Update the version of rustc, open a PR and assign a developer.
  update:
    runs-on: ubuntu-latest
    steps:
      - name: Check out the repo
        uses: actions/checkout@v2

      - name: Install cargo-outdated
        run: cargo install --locked cargo-outdated

      - name: Update dependencies
        run: |
          # Update the rustc version
          NIGHTLY_VERSION="nightly-$(
            curl -s https://rust-lang.github.io/rustup-components-history/x86_64-unknown-linux-gnu/rustc-dev
          )"
          echo "The latest usable version of rustc is $NIGHTLY_VERSION"
          sed -i 's/^channel\s*=.*$/channel = "'"$NIGHTLY_VERSION"'"/' rust-toolchain
          declare -a DEVELOPERS=(Aurel300 fpoli vakaras)
          UPDATE_NUMBER=$(( $(date +%m) * 2 + ($(date +%d) / 15) - 2 ))
          DEVELOPER="${DEVELOPERS[ $UPDATE_NUMBER % ${#DEVELOPERS[@]} ]}"
          echo "The assigned developer is $DEVELOPER"
          echo "NIGHTLY_VERSION=$NIGHTLY_VERSION" &gt;&gt; $GITHUB_ENV
          echo "DEVELOPER=$DEVELOPER" &gt;&gt; $GITHUB_ENV
          # Find outdated dependencies
          echo 'OUTDATED_DEPENDENCIES=&lt;&lt;EOF' &gt;&gt; $GITHUB_ENV
          cargo outdated --root-deps-only --workspace &gt;&gt; $GITHUB_ENV
          echo 'EOF' &gt;&gt; $GITHUB_ENV

      - name: Open a pull request
        uses: peter-evans/create-pull-request@v3
        with:
          # Use viper-admin's token to workaround a restriction of GitHub.
          # See: https://github.com/peter-evans/create-pull-request/issues/48
          token: ${{ secrets.VIPER_ADMIN_TOKEN }}
          commit-message: Update rustc to ${{ env.NIGHTLY_VERSION }}
          title: Update dependencies (${{ env.NIGHTLY_VERSION }})
          branch: auto-update-deps
          delete-branch: true
          assignees: ${{ env.DEVELOPER }}
          body: |
            * [x] Update rustc version to `${{ env.NIGHTLY_VERSION }}`.
            * [ ] Manualy update outdated dependencies (see the list below).
            * [ ] Manualy run `cargo update`.
            
            &lt;details&gt;&lt;summary&gt;List of direct outdated dependencies:&lt;/summary&gt;

            ```
            $ cargo outdated --root-deps-only --workspace
            
            ${{ env.OUTDATED_DEPENDENCIES }}
            ```
            &lt;/details&gt;

            @${{ env.DEVELOPER }} could you take care of this?
</code></pre>
</details>

**Comment by fpoli** [comment, marked as answer]
<p>I found the issue, instead of</p>
<pre><code class="lang-auto">echo 'OUTDATED_DEPENDENCIES=&lt;&lt;EOF' &gt;&gt; $GITHUB_ENV
</code></pre>
<p>it should be</p>
<pre><code class="lang-auto">echo 'OUTDATED_DEPENDENCIES&lt;&lt;EOF' &gt;&gt; $GITHUB_ENV
</code></pre>

**Reply by faizanakram99** [reply]
whats the difference? both are the same

**Comment by epignatelli** [comment]
It always strikes me how low level this syntax is for such a high level task like a GitHub action.
